<img src="https://i.ibb.co/93sNcdhp/Logo-pmat-color.png" align="right" width="150">

<br>

# **Introducción**

En estadística inferencial, el contraste de hipótesis permite tomar decisiones fundamentadas a partir de información muestral. En muchos contextos reales, el interés no se centra en describir una sola población, sino en **comparar dos poblaciones o dos grupos** con el fin de determinar si existen diferencias significativas entre ellas.

En este laboratorio se aborda el contraste de hipótesis en situaciones de comparación, considerando dos parámetros de interés ampliamente utilizados: la **diferencia de medias poblacionales** y la **diferencia de proporciones poblacionales**. Estos procedimientos permiten evaluar, por ejemplo, si dos grupos presentan rendimientos distintos, si un tratamiento produce efectos diferentes respecto de otro o si la proporción de ocurrencia de un evento varía entre dos poblaciones.

Si bien el tipo de variable y el estadístico de prueba pueden cambiar según el caso, ambos contrastes comparten una **misma estructura inferencial**: formulación de hipótesis nula y alternativa, selección del estadístico de prueba adecuado, cálculo del valor $p$ y toma de decisiones basada en un nivel de significancia previamente definido.

El objetivo de este laboratorio es que comprendas y apliques el contraste de hipótesis para la diferencia de medias y la diferencia de proporciones, interpretando correctamente los resultados obtenidos y relacionándolos con situaciones reales provenientes de ámbitos como la educación, la salud, la industria y las ciencias sociales.


## **Objetivos de aprendizaje**

Al finalizar este laboratorio, serás capaz de:

- Comprender el procedimiento de **contraste de hipótesis aplicado a la comparación entre dos poblaciones**, considerando la **diferencia de medias** y la **diferencia de proporciones** como parámetros de interés.
- Formular correctamente la **hipótesis nula $(H_{0})$** y la **hipótesis alternativa $(H_{1})$** en contextos de muestras independientes.
- Seleccionar y calcular el **estadístico de prueba** correspondiente ($z$ o $t$), de acuerdo con la información disponible y el tipo de parámetro analizado.
- Calcular e interpretar el **valor $p$** en relación con un nivel de significancia $\alpha$.
- Tomar decisiones fundamentadas respecto del rechazo o no rechazo de la hipótesis nula, considerando tanto el criterio estadístico como el **contexto del problema**.
- Aplicar el contraste de hipótesis para la diferencia de medias y la diferencia de proporciones en **situaciones reales** provenientes de ámbitos científicos, educativos, comerciales, de salud y tecnológicos.


## **Recordatorio clave**

### Hipótesis estadísticas en la comparación de dos poblaciones

En problemas de comparación entre dos poblaciones independientes, el contraste de hipótesis se centra en evaluar si existe una diferencia significativa entre los parámetros de interés.

- **Hipótesis nula $(H_{0})$**: no existe diferencia entre las poblaciones.
- **Hipótesis alternativa $(H_{1})$**: existe una diferencia entre las poblaciones, la cual puede ser **bilateral** o **unilateral**, según el contexto del problema.
<br>

### Contraste para la diferencia de medias poblacionales

**Hipótesis generales:**

$$H_{0}: \ \mu_1 = \mu_2 \quad \text{o} \quad H_{0}:\mu_1 \leq \mu_2 \quad \text{o} \quad H_{0}:\mu_1 \geq \mu_2$$

$$H_{1}: \ \mu_1 \neq \mu_2 \quad \text{o} \quad H_{1}:\mu_1 > \mu_2 \quad \text{o} \quad H_{1}:\mu_1 < \mu_2$$
<br>

#### Estadístico de prueba

En este laboratorio, para contrastar hipótesis sobre la diferencia entre dos medias de muestras independientes, se utilizará la **prueba $t$ de Welch**.

El estadístico de prueba está dado por:

$$t = \frac{\bar{X}_1 - \bar{X}_2}{\sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}}}$$

con grados de libertad aproximados:

$$gl =
\frac{
\left( \frac{s_1^2}{n_1} + \frac{s_2^2}{n_2} \right)^2
}{
\frac{\left( \frac{s_1^2}{n_1} \right)^2}{n_1 - 1} +
\frac{\left( \frac{s_2^2}{n_2} \right)^2}{n_2 - 1}
}$$

**Nota:** En algunos contextos, cuando se puede asumir igualdad de varianzas poblacionales, puede utilizarse la prueba t de Student. Sin embargo, en este laboratorio trabajaremos únicamente con la prueba t de Welch.
<br>

### Contraste para la diferencia de proporciones poblacionales

**Hipótesis generales:**

$$H_{0}: \ p_1 = p_2 \quad \text{o} \quad H_{0}:p_1 \leq p_2 \quad \text{o} \quad H_{0}:p_1 \geq p_2$$

$$H_{1}: \ p_1 \neq p_2 \quad \text{o} \quad H_{1}:p_1 > p_2 \quad \text{o} \quad H_{1}:p_1 < p_2$$

#### Estadístico de prueba

Las proporciones muestrales se definen como:

$$\hat{p}_1 = \frac{x_1}{n_1},
\qquad
\hat{p}_2 = \frac{x_2}{n_2}$$

Bajo la hipótesis nula, se utiliza la **proporción combinada**:

$$\hat{p} = \frac{x_1 + x_2}{n_1 + n_2}$$

El estadístico de prueba es:

$$z = \frac{\hat{p}_1 - \hat{p}_2}
{\sqrt{\hat{p}(1 - \hat{p})\left(\frac{1}{n_1} + \frac{1}{n_2}\right)}}$$
<br>

### Regla de decisión

Para ambos contrastes:

- Si **valor $p$ $<$ $\alpha$** → se **rechaza $H_{0}$**.
- Si **valor $p$ $≥$ $\alpha$** → **no se rechaza $H_{0}$**.

donde $\alpha$ corresponde al **nivel de significancia** previamente establecido.
<br>

### Errores tipo I y tipo II

En el contraste de hipótesis para la comparación entre dos poblaciones, pueden cometerse dos tipos de errores en la toma de decisiones:

- **Error tipo I $(\alpha)$**: rechazar la hipótesis nula cuando en realidad **no existe diferencia** entre las poblaciones (por ejemplo, $\mu_1 = \mu_2$ o $p_1 = p_2$).
- **Error tipo II $(\beta)$**: no rechazar la hipótesis nula cuando en realidad **sí existe una diferencia** entre las poblaciones (por ejemplo, $\mu_1 \neq \mu_2$ o $p_1 \neq p_2$).

El valor de $\alpha$ corresponde al **nivel de significancia** fijado previamente y representa la probabilidad de cometer un error tipo I. El complemento $1 - \beta$ se denomina **poder de la prueba** y mide la capacidad del contraste para detectar una diferencia real cuando esta existe.

*Nota:* Aumentar el **tamaño muestral** contribuye a disminuir la probabilidad de cometer errores tipo I y tipo II, y a incrementar el poder de la prueba, mejorando la calidad de la inferencia estadística.


In [ ]:
# @title
%%html
<div style="border: 4px solid #006d77; border-radius: 0px; overflow: hidden; font-family: sans-serif; max-width: 800px; margin-bottom: 20px;">

  <div style="background-color: #000000; color: #ffc107; padding: 10px 15px; font-weight: bold; font-size: 16px; display: flex; align-items: center;">
    <span style="margin-right: 10px; font-size: 18px;">💡</span>
    <span>Prueba de hipótesis para la diferencia de medias</span>
  </div>

  <div style="background-color: #ffffff; padding: 20px; display: flex; align-items: center; justify-content: space-between;">

    <div style="color: #000000; flex: 1; padding-right: 30px; font-size: 14px; line-height: 1.5;">
      ¿Te has preguntado cómo comparar el rendimiento, tiempo de respuesta o desempeño promedio entre dos grupos?
      <b>Escanea o haz clic sobre el código QR</b> para conocer cómo se aplican las pruebas de hipótesis para la
      diferencia de medias y cómo apoyan la toma de decisiones basada en datos.
      </div>

    <div style="flex-shrink: 0;">
      <a href="https://duoc.h5p.com/content/1292769935177997348" target="_blank">
        <img src="https://i.ibb.co/6JvYN53t/PH-dif-medias.png"
             alt="QR Code"
             style="width: 100px; height: 100px; border: none;">
      </a>
    </div>

  </div>

  <div style="background-color: #000000; color: #ffc107; padding: 8px 15px; font-weight: bold; font-size: 16px;">
    Presentación interactiva H5P
  </div>

</div>


# **Actividad 1 — Productividad en dos turnos de trabajo**

Una empresa está interesada en evaluar si existe una diferencia en la productividad promedio (unidades producidas por hora) entre los trabajadores del *turno diurno* y el *turno nocturno*. Para ello, se seleccionaron dos muestras aleatorias independientes de trabajadores de ambos turnos.

Con esta información, la gerencia busca identificar si el horario de trabajo influye en el desempeño y, en caso de haber diferencias, analizar posibles ajustes en la organización de los turnos.
A partir de las muestras, se obtuvo la siguiente información:
- Turno día: $n_{1} = 30$, media = 52, $s_{1} = 5$.  
- Turno noche: $n_{2} = 28$, media = 49, $s_{2} = 6$.

Considerando un nivel de significancia de $\alpha = 0,05$.


a.	Formule la hipótesis nula y la hipótesis alternativa.

**Hipótesis**  
- $H_{0}: \mu_{diurno} = \mu_{nocturno}$  
- $H_{1}: \mu_{diurno} \neq \mu_{nocturno}$

En lenguaje verbal productividad promedio
- $H_{0}:$ productividad promedio turno diurno $=$ productividad promedio turno nocturno
- $H_{1}:$ productividad promedio turno diurno $\neq$ productividad promedio turno nocturno

b.	Calcule el estadístico de prueba.

In [1]:
import numpy as np
from scipy.stats import t

# Datos
n1, x1, s1 = 30, 52, 5
n2, x2, s2 = 28, 49, 6
alpha = 0.05

# Error estándar
se = np.sqrt(s1**2/n1 + s2**2/n2)

# t de Welch (no asume varianzas iguales)
t_turnos = (x1 - x2) / se

# Grados de libertad
gl = (se**4) / (((s1**2/n1)**2)/(n1-1) + ((s2**2/n2)**2)/(n2-1))


print(f"Estadístico de prueba t = {t_turnos:.3f}")

Estadístico de prueba t = 2.061


c.	Determine el valor $p$.

In [4]:
# Cálculo valor p
p_valor = 2 * (1 - t.cdf(abs(t_turnos), gl))
print(f"Valor p = {p_valor:.4f}")

Valor p = 0.0443


d.	Compare el valor $p$ con el nivel de significancia y establezca la decisión del contraste.

In [5]:
# Comparación y decisión

if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

Rechazar H0


e.	Redacte una conclusión final en el contexto del problema. ¿Existe evidencia suficiente para afirmar que la productividad promedio difiere entre el turno diurno y el turno nocturno?

El contraste de hipótesis arroja un estadístico $t$ = 2,061 con un p-valor = 0,0443.
Dado que el valor $p$ es ligeramente menor que el nivel de significancia habitual $(\alpha = 0,05)$, se rechaza $H_{0}$ y se concluye que sí existen diferencias significativas en la productividad entre los dos turnos. En otras palabras, la evidencia sugiere que el turno diurno y el nocturno no tienen el mismo rendimiento promedio.

# **Actividad 2 — Rendimiento académico en dos grupos**

Un profesor aplicó la misma prueba en dos secciones de la asignatura, pero las secciones tuvieron metodologías de trabajo distintas: el Grupo 1 realizó controles semanales y el Grupo 2 hizo dos controles acumulativos. Aunque el contenido evaluado fue idéntico, el equipo docente quiere saber si estas diferencias operativas se reflejan en el rendimiento promedio.
Los puntajes obtenidos por cada grupo son los siguientes:

- Grupo 1: $n_{1}$ = 25, media = 68, $s_{1}$ = 8  
- Grupo 2: $n_{2}$ = 22, media = 72, $s_{2}$ = 7  

Realiza un test de hipótesis para diferencia de medias con un nivel de significancia $\alpha = 0.05$.  


a.	Formule la hipótesis nula y la hipótesis alternativa.

**Hipótesis**

- $H_{0} : \mu_{grupo1} = \mu_{grupo2}$

- $H_{1} : \mu_{grupo1} \neq \mu_{grupo2}$

En lenguaje verbal

- $H_{0} :$ Rendimiento promedio grupo 1 $=$ Rendimiento promedio grupo 2

- $H_{1} :$ Rendimiento promedio grupo 1 $\neq$ Rendimiento promedio grupo 2

b.	Calcule el estadístico de prueba.

In [6]:
import numpy as np
from scipy.stats import t

# Datos
n1, x1, s1 = 25, 68, 8
n2, x2, s2 = 22, 72, 7
alpha = 0.05

# Error estándar
se = np.sqrt(s1**2/n1 + s2**2/n2)

# t de Welch (no asume varianzas iguales)
t_rendimiento = (x1 - x2) / se

# Grados de libertad
gl = (se**4) / (((s1**2/n1)**2)/(n1-1) + ((s2**2/n2)**2)/(n2-1))

print(f"Estadístico de prueba t = {t_rendimiento:.3f}")

Estadístico de prueba t = -1.828


c.	Determine el valor $p$.

In [ ]:
# Cálculo valor p
p_valor = 2 * (1 - t.cdf(abs(t_rendimiento), gl))
print(f"Valor p = {p_valor:.4f}")

Valor p = 0.0742


d.	Compare el valor $p$ con el nivel de significancia y establezca la decisión del contraste.

In [7]:
# Comparación y decisión

if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

Rechazar H0


e.	Redacte una conclusión en el contexto del problema. ¿Existe evidencia suficiente para afirmar que el rendimiento promedio difiere entre los grupos que trabajaron con metodologías distintas?

Con $\alpha = 0,05$, el valor $p$ = 0,0742 es mayor que 0,05. por lo tanto no se rechaza $H_{0}$. Aunque el grupo 2 obtuvo un puntaje promedio mayor que el grupo 1, la evidencia no es suficiente al 5% para afirmar que los promedios poblacionales difieran.

# **Actividad 3 — Tratamiento médico vs placebo**

En un ensayo clínico controlado, dos grupos de pacientes con hipertensión leve reciben intervenciones distintas durante dos semanas. El grupo tratamiento recibe un nuevo fármaco, mientras que el grupo placebo recibe una pastilla inerte con el mismo aspecto para mantener el cegamiento. Los médicos quieren evaluar si el tratamiento logra una mayor reducción promedio de la presión arterial que el placebo.

Al finalizar las dos semanas, se obtuvo la siguiente información resumen sobre la reducción de presión arterial, medida en mmHg:
- Tratamiento: $n_{1}$ = 20, media = 12 mmHg, $s_{1}$ = 4  
- Placebo: $n_{2}$ = 18, media = 9 mmHg, $s_{2}$ = 3  

Determina si el nuevo tratamiento logra una mayor reducción promedio que el placebo. Utiliza un nivel de significancia del 1%.

1. Formule las hipótesis

**Hipótesis**

- $H_{0} : \mu_{tratamiento} \leq \mu_{placebo}$

- $H_{1} : \mu_{tratamiento} > \mu_{placebo}$

En lenguaje verbal
- $H_{0} :$ Reducción promedio grupo tratamiento $\leq$ Reducción promedio grupo placebo

- $H_{1} :$ Reducción promedio grupo tratamiento $>$ Reducción promedio grupo placebo


2. Construya el contraste de hipótesis

In [ ]:
import scipy.stats as st

# Datos
n1, x1, s1 = 20, 12, 4
n2, x2, s2 = 18, 9, 3
alpha = 0.01

# Error estándar
se = np.sqrt(s1**2/n1 + s2**2/n2)

# t de Welch (no asume varianzas iguales)
t = (x1 - x2) / se

# Grados de libertad
gl = (se**4) / (((s1**2/n1)**2)/(n1-1) + ((s2**2/n2)**2)/(n2-1))

# Cálculo valor p
p_valor = 1 - st.t.cdf(t, gl)

print(f"t = {t:.3f}, p-valor = {p_valor:.4f}")

t = 2.631, p-valor = 0.0063


In [ ]:
# Comparación y decisión

if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

Rechazar H0


3. Interprete los resultados obtenidos en el contexto del problema

Con $\alpha = 0,01$, el valor $p$ = 0,0063 es menor que 0,01. por lo tanto, se rechaza $H_{0}$. Hay evidencia estadística de que el tratamiento reduce en promedio más la presión arterial que el placebo.

# **Actividad 4 — Eficiencia energética en motores eléctricos**

Una empresa de manufactura reemplazó recientemente los motores de una de sus líneas de producción por un nuevo modelo con mayor eficiencia energética. El departamento de ingeniería desea evaluar si el consumo promedio de energía eléctrica, medido en kWh por hora de operación, ha disminuido con el nuevo modelo en comparación con el modelo anterior.

Para ello, se dispone de un conjunto de datos que contiene mediciones de consumo energético correspondientes a motores del modelo antiguo y del nuevo modelo.

Descargue el set de datos **<a href="https://docs.google.com/spreadsheets/d/154B6V7Ug2Kcd_4fPGTPp1--K-VjinZAV/export?format=xlsx">aquí</a>.**

A partir de esta información, realice un contraste de hipótesis para determinar si el consumo promedio del nuevo modelo es menor que el del modelo antiguo, utilizando un nivel de significancia del 5%. Para ello:


a.	Formule las hipótesis del problema.

**Hipótesis**

- $H_{0} : \mu_{nuevo} \geq \mu_{antiguo}$

- $H_{1} : \mu_{nuevo} < \mu_{antiguo}$

En lenguaje verbal
- $H_{0} :$ Consumo promedio nuevo $\geq$ Consumo promedio antiguo

- $H_{1} :$ Consumo promedio nuevo $<$ Consumo promedio antiguo


b.	Cargue la base de datos al entorno de trabajo.

In [ ]:
import pandas as pd
df = pd.read_excel("G5_A4_consumo_motores.xlsx")
df.head(3)

,antiguo_kwh,nuevo_kwh
0,14.692605,15.886848
1,13.432072,14.747364
2,15.192436,12.996411


c.	Calcule los estadísticos necesarios para realizar el contraste de hipótesis.

In [ ]:
import numpy as np
from scipy.stats import t
import scipy.stats as st

# Datos
# Extraer muestras desde el df
x1 = df["antiguo_kwh"]  # modelo antiguo
x2 = df["nuevo_kwh"]    # modelo nuevo

# Estadísticos muestrales necesarios ---
n1, n2 = len(x1), len(x2)
x1_bar, x2_bar = x1.mean(), x2.mean()
s1, s2 = x1.std(ddof=1), x2.std(ddof=1)  # sd muestral (ddof=1)

print(f"Antiguo -> n={n1}, media={x1_bar:.1f}, s={s1:.1f}")
print(f"Nuevo   -> n={n2}, media={x2_bar:.1f}, s={s2:.1f}")

Antiguo -> n=20, media=15.8, s=1.2
Nuevo   -> n=20, media=15.2, s=1.0


d.	Realice el contraste para la diferencia de medias.

In [ ]:
# Nivel de significancia
alpha = 0.05

# Error estándar
se = np.sqrt(s1**2/n1 + s2**2/n2)

# t de Welch (no asume varianzas iguales)
t_motores = (x2_bar - x1_bar) / se

# Grados de libertad
gl = (se**4) / (((s1**2/n1)**2)/(n1-1) + ((s2**2/n2)**2)/(n2-1))

print(f"Estadístico de prueba t = {t_motores:.3f}")

Estadístico de prueba t = -1.718


In [ ]:
# Cálculo valor p
p_valor = st.t.cdf(t_motores, df=gl)
print(f"Valor p = {p_valor:.4f}")

Valor p = 0.0471


In [ ]:
# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

Rechazar H0


e.	Interprete los resultados obtenidos en el contexto del problema.

Con $\alpha = 0,05$, el valor $p$ = 0,0471 es menor que 0,05. por lo tanto, se rechaza $H_{0}$. Existe evidencia estadística suficiente al 5 % para afirmar que el nuevo modelo consume menos energía en promedio que el modelo antiguo.

# **Actividad 5 - Contenido de vitamina A en la sangre**

Se desea comparar el contenido de vitamina A en la sangre entre trabajadores que se desempeñan a nivel del mar y trabajadores que se desempeñan en altura.

Para ello, se dispone de una base de datos con mediciones del contenido de vitamina A en ambos grupos. A partir de esta información, determine si el contenido medio de vitamina A en la sangre es mayor en los trabajadores en altura que en los trabajadores a nivel del mar, utilizando un nivel de significancia del 5%.

Descargue la base de datos **<a href="https://docs.google.com/spreadsheets/d/1gfcxetMoDk3Fg_aCFwD87_E5B3n8Zv9j/export?format=xlsx">aquí</a>.** Luego, realice lo siguiente:

a. Formule las hipótesis del problema.

**Hipótesis**

- $H_{0} : \mu_{altura} \leq \mu_{mar}$

- $H_{1} : \mu_{altura} > \mu_{mar}$

En lenguaje verbal
- $H_{0} :$ Vitamina promedio en altura $\leq$ Vitamina promedio a nivel del mar

- $H_{1} :$ Vitamina promedio en altura $>$ Vitamina promedio a nivel del mar

b. Cargue la base de datos al entorno de trabajo.

In [ ]:
import pandas as pd

df_5 = pd.read_excel("G5_A5_vitamina_A.xlsx")
df_5.head()

,vitamina_mar,vitamina_altura
0,25.2,43.7
1,30.4,62.6
2,46.9,61.6
3,51.0,74.8
4,46.4,36.8


c. Calcule los estadísticos necesarios para realizar el contraste de hipótesis.

In [ ]:
import numpy as np
from scipy.stats import t
import scipy.stats as st

# Datos
# Extraer muestras desde el df
x1 = df_5["vitamina_altura"]  # trabajadores en altura
x2 = df_5["vitamina_mar"]    # trabajadores a nivel del mar

# Estadísticos muestrales necesarios ---
n1, n2 = len(x1), len(x2)
x1_bar, x2_bar = x1.mean(), x2.mean()
s1, s2 = x1.std(ddof=1), x2.std(ddof=1)  # sd muestral (ddof=1)

print(f"Trabajadores en altura -> n={n1}, media={x1_bar:.1f}, s={s1:.1f}")
print(f"Trabajadores a nivel del mar   -> n={n2}, media={x2_bar:.1f}, s={s2:.1f}")

Trabajadores en altura -> n=24, media=55.7, s=10.9
Trabajadores a nivel del mar   -> n=24, media=39.4, s=8.1


d. Realice el contraste para la diferencia de medias.

In [ ]:
# Nivel de significancia
alpha = 0.05

# Error estándar
se = np.sqrt(s1**2/n1 + s2**2/n2)

# t de Welch (no asume varianzas iguales)
t_vitamina = (x2_bar - x1_bar) / se

# Grados de libertad
gl = (se**4) / (((s1**2/n1)**2)/(n1-1) + ((s2**2/n2)**2)/(n2-1))

print(f"Estadístico de prueba t = {t_vitamina:.3f}")

Estadístico de prueba t = -5.882


In [ ]:
# Cálculo valor p
p_valor = 1 - t.cdf(t_vitamina, gl)
print(f"Valor p = {p_valor:.4f}")

Valor p = 1.0000


In [ ]:
# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

No rechazar H0


e. Interprete los resultados obtenidos en el contexto del problema.

Con $\alpha = 0,05$, el valor $p$ = 1,0000 es mayor que 0,05, por lo tanto, no se rechaza $H_{0}$. No existe evidencia estadística para afirmar que contenido medio de vitamina A en la sangre es mayor en los trabajadores que se desempeñan en altura respecto a los que trabajan a nivel del mar.

# **Contraste de hipótesis para la diferencia de proporciones**

## **Transición: de la diferencia de medias a la diferencia de proporciones**

Hasta este punto, has trabajado el contraste de hipótesis para la **diferencia de medias poblacionales**, el cual se utiliza cuando el interés está en comparar el comportamiento promedio de una variable cuantitativa entre dos poblaciones independientes.

En muchos contextos reales, sin embargo, la comparación no se centra en valores promedio, sino en la **proporción de individuos que presentan una determinada característica** en cada población, como la ocurrencia de un evento, la aprobación de un curso o la adopción de una conducta específica. En estos casos, la variable de interés es de tipo categórica dicotómica y el parámetro que se compara es la **diferencia de proporciones poblacionales**.

A pesar de estas diferencias, ambos contrastes comparten una **misma estructura inferencial**: formulación de hipótesis nula y alternativa, selección del estadístico de prueba adecuado, cálculo del valor p y toma de decisiones basada en un nivel de significancia previamente definido. En la siguiente sección se abordará el contraste de hipótesis para la diferencia de proporciones, destacando tanto sus similitudes como sus particularidades respecto del caso de medias.

In [ ]:
# @title
%%html
<div style="border: 4px solid #006d77; border-radius: 0px; overflow: hidden; font-family: sans-serif; max-width: 800px; margin-bottom: 20px;">

  <div style="background-color: #000000; color: #ffc107; padding: 10px 15px; font-weight: bold; font-size: 16px; display: flex; align-items: center;">
    <span style="margin-right: 10px; font-size: 18px;">💡</span>
    <span>Prueba de hipótesis para la diferencia de proporciones</span>
  </div>

  <div style="background-color: #ffffff; padding: 20px; display: flex; align-items: center; justify-content: space-between;">

    <div style="color: #000000; flex: 1; padding-right: 30px; font-size: 14px; line-height: 1.5;">
      ¿Te has preguntado cómo comparar la frecuencia de un resultado entre dos grupos,
      como la proporción de éxito, error o preferencia? <b>Escanea o haz clic sobre el código QR</b> para conocer cómo se aplican las pruebas de hipótesis
      para la diferencia de proporciones y cómo apoyan la comparación de resultados en contextos de análisis de datos.
      </div>

    <div style="flex-shrink: 0;">
      <a href="https://duoc.h5p.com/content/1292769936935048558" target="_blank">
        <img src="https://i.ibb.co/bR6dzpK7/PH-dif-proporciones.png"
             alt="QR Code"
             style="width: 100px; height: 100px; border: none;">
      </a>
    </div>

  </div>

  <div style="background-color: #000000; color: #ffc107; padding: 8px 15px; font-weight: bold; font-size: 16px;">
    Presentación interactiva H5P
  </div>

</div>


# **Actividad 6 — Efectividad de clases híbridas**

Un Instituto Profesional detectó que ciertas asignaturas presentan altas tasas de reprobación y decidió probar una estrategia de clases híbridas (combinación de sesiones presenciales con actividades en línea sincrónicas/asíncronas) para mejorar el aprendizaje y la retención.

En un semestre se trabajó así:
* Grupo híbrido: 120 estudiantes inscritos, de los cuales 78 aprobaron.
* Grupo presencial tradicional: 150 estudiantes inscritos, de los cuales 85 aprobaron.

Como la implementación de clases híbridas implica cambios en la planificación docente, uso de plataformas y capacitación del profesorado, la universidad quiere saber si realmente hay diferencia en la tasa de aprobación entre ambas modalidades antes de escalar el modelo a más asignaturas.

A un nivel de significancia del 5 %, ¿la proporción de estudiantes que aprueban es distinta entre la modalidad híbrida y la presencial?

a.	Formule la hipótesis nula y la hipótesis alternativa.

**Hipótesis**

- $H_{0} : p_{híbrido} = p_{presencial}$
- $H_{1} : p_{híbrido} \neq p_{presencial}$

El lenguaje verbal
- $H_{0} :$ Proporción de estudiantes formato híbrido $=$ Proporción de estudiantes formato presencial
- $H_{1} :$ Proporción de estudiantes formato híbrido $\neq$ Proporción de estudiantes formato presencial

b.	Calcule el estadístico de prueba.

In [ ]:
import numpy as np
from scipy.stats import norm
import scipy.stats as st

# Datos
n1, x1 = 120, 78 # Grupo híbrido
n2, x2 = 150, 85 # Grupo presencial
alpha = 0.05

# Proporciones muestrales
p1 = x1 / n1
p2 = x2 / n2

# Proporción combinada
p = (x1 + x2) / (n1 + n2)

# Estadístico de Prueba
z_clases = (p1 - p2) / np.sqrt(p*(1-p)*(1/n1 + 1/n2))

print(f"Estadístico de prueba z = {z_clases:.3f}")

Estadístico de prueba z = 1.391


c.	Determine el valor $p$.

In [ ]:
# Cálculo valor p
p_valor = 2 * (1 - st.norm.cdf(abs(z_clases)))

print(f"Valor p = {p_valor:.4f}")

Valor p = 0.1642


d.	Compare el valor $p$ con el nivel de significancia y establezca la decisión del contraste.

In [ ]:
# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

No rechazar H0


e.	Interprete los resultados obtenidos en el contexto del problema.

Con un nivel de significancia $\alpha = 0.05$, y un valor $p$ = 0,1642, dado que el valor $p$ es mayor que el valor de $\alpha$, no se rechaza $H_{0}$. Con los datos disponibles y usando un nivel de significancia del 5 %, no se puede afirmar que la modalidad híbrida tenga una tasa de aprobación diferente a la presencial.

# **Actividad 7 — Campaña de vacunación en dos centros de salud**

El Ministerio de Salud quiere evaluar si vale la pena invertir en campañas de vacunación más intensivas para aumentar la cobertura contra la influenza en la atención primaria.

Para ello, se implementaron dos estrategias distintas en dos centros de salud:

*	**Centro A:** estrategia reforzada, que incluyó visitas domiciliarias, difusión en redes sociales y llamados telefónicos de recordatorio.
*	**Centro B:** estrategia tradicional, basada en la invitación presencial cuando el paciente asistía al centro de salud.

Al finalizar la campaña, se registró la siguiente información:

*	Centro A: 200 pacientes elegibles, de los cuales 170 se vacunaron.
*	Centro B: 180 pacientes elegibles, de los cuales 138 se vacunaron.

Dado que implementar la estrategia reforzada implica mayor costo y carga de trabajo, la autoridad sanitaria exige un nivel de significancia del 1 % para concluir que realmente esta estrategia logra una proporción de vacunación mayor que la del método tradicional, antes de decidir escalarla a otros centros.

¿Hay evidencia al 1 % para concluir que el Centro A tiene una proporción mayor de vacunación que el Centro B?

a.	Formule la hipótesis nula y la hipótesis alternativa.

**Hipótesis**

- $H_{0} : p_{centroA} \leq p_{centroB}$
- $H_{1} : p_{centroA} > p_{centroB}$

En lenguaje verbal
- $H_{0} :$ Proporción vacunados centro A $\leq$ Proporción vacunados centro B
- $H_{1} :$ Proporción vacunados centro A $>$ Proporción vacunados centro B


b.	Calcule el estadístico de prueba.

In [ ]:
import numpy as np
from scipy.stats import norm
import scipy.stats as st

# Datos
n1, x1 = 200, 170  # Centro A
n2, x2 = 180, 138  # Centro B
alpha =  0.01

# Proporciones muestrales
p1 = x1 / n1
p2 = x2 / n2

# Proporción combinada
p_comb = (x1 + x2) / (n1 + n2)

# Estadístico de prueba
z_vacunas = (p1 - p2) / np.sqrt(p_comb * (1 - p_comb) * (1/n1 + 1/n2))

print(f"p1 = {p1*100:.2f}%, p2 = {p2*100:.2f}%")
print(f"Estadístico de prueba z = {z_vacunas:.3f}")

p1 = 85.00%, p2 = 76.67%
Estadístico de prueba z = 2.070


c.	Determine el valor $p$.

In [ ]:
# Cálculo valor p
p_valor = 1 - st.norm.cdf(z_vacunas)

print(f"Valor p = {p_valor:.4f}")

Valor p = 0.0192


d.	Compare el valor $p$ con el nivel de significancia y establezca la decisión del contraste.

In [ ]:
# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

No rechazar H0


3. Interprete el resultado

Con $\alpha = 0.01$, el valor $p$ obtenido (0.0192) es mayor que $\alpha$, por lo tanto NO se rechaza $H_{0}$.

Esto significa que, aunque el Centro A muestra una proporción de vacunación un poco más alta que el Centro B (85 % vs 76.67 %), la evidencia estadística no es suficiente al nivel de significancia del 1 % para afirmar que realmente es mayor.

# **Actividad 8 — Adopción de software de gestión**

Una empresa tecnológica renovó su sistema interno de gestión y capacitó de forma distinta a dos áreas: *Ventas* participó en talleres prácticos presenciales, donde se simulaban casos reales de uso del sistema y *Soporte* solo recibió manuales digitales y videos explicativos, sin instancias prácticas guiadas.

La gerencia sospecha que este enfoque más “pasivo” en Soporte podría haber generado una menor adopción del nuevo sistema en comparación con Ventas y quiere verificarlo con datos.

Al cabo de un mes se registró lo siguiente:

*	Ventas: 90 usuarios, de los cuales 72 adoptaron el sistema.
*	Soporte: 110 usuarios, de los cuales 70 adoptaron el sistema.

Se desea evaluar, con un nivel de significancia del 10 %, si la proporción de adopción en Soporte es menor que en Ventas.

a.	Plantee la hipótesis nula y la hipótesis alternativa para este problema.

**Hipótesis**

- $H_{0} : p_{soporte} \geq p_{ventas}$
- $H_{1} : p_{soporte} < p_{ventas}$

En lenguaje verbal
- $H_{0} :$ Proporción de adopción en grupo soporte $\geq$ Proporción de adopción en grupo ventas
- $H_{1} :$ Proporción de adopción en grupo soporte $<$ Proporción de adopción en grupo ventas

b.	Realice el contraste de hipótesis para la diferencia de proporciones en Python.

In [ ]:
import numpy as np
from scipy.stats import norm
import scipy.stats as st

# Datos
n1, x1 = 110, 70  # Soporte
n2, x2 = 90, 72   # Ventas
alpha = 0.10

# Proporciones muestrales
p1 = x1 / n1
p2 = x2 / n2

# Proporción combinada
p_comb = (x1 + x2) / (n1 + n2)

# Estadístico de prueba
se = np.sqrt(p_comb * (1 - p_comb) * (1/n1 + 1/n2))
z_software = (p1 - p2) / se

# Cálculo valor p
p_valor = st.norm.cdf(z_software)

print(f"p1 (Soporte) = {p1*100:.2f}%, p2 (Ventas) = {p2*100:.2f}%")
print(f"z = {z_software:.3f}, p-valor = {p_valor:.4f}")

p1 (Soporte) = 63.64%, p2 (Ventas) = 80.00%
z = -2.537, p-valor = 0.0056


c.	A partir del valor $p$ obtenido, establezca la decisión del contraste.

In [ ]:
# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

Rechazar H0


d.	Explique qué concluye la empresa respecto de la adopción del software en ambas áreas.

La proporción de adopción del software es mayor en Ventas (80 %) que en Soporte (63,64 %) y la diferencia es estadísticamente significativa con un nivel de significancia del 10 %.

Por tanto, se rechaza $H_{0}$ y se concluye que la proporción de adopción del software en Soporte es significativamente menor que en Ventas.

# **Actividad 9 — Compras online antes y después de un rediseño web**

Una tienda en línea detectó que su tasa de conversión (porcentaje de visitas que terminan en una compra) era más baja de lo deseado. Para mejorar la experiencia de usuario y aumentar las ventas, el equipo implementó un rediseño del sitio web, incorporando mejoras como la simplificación del proceso de compra, la reorganización del catálogo, una mejor velocidad de carga y recomendaciones personalizadas.

Para evaluar si el rediseño realmente funcionó, se dispone de un set de datos con información de visitas registradas antes y después del rediseño. El set de datos contiene las variables antes_rediseño y despues_rediseño, donde:

*	0 indica que la visita no terminó en compra.
*	1 indica que la visita sí terminó en compra.

A partir de esta información, determine si la proporción de visitas que terminan en compra es mayor después del rediseño que antes, utilizando un nivel de significancia del 5 %.

Descargue el set de datos **<a href="https://drive.google.com/uc?export=download&id=1N3DAv9JDKvFTCYbJlL3Zq7WGrlxmceBI">aquí</a>**.

a.	Formule las hipótesis del problema.

**Hipótesis**

- $H_{0} : p_{despues} \leq p_{antes}$
- $H_{1} : p_{despues} > p_{antes}$

En lenguaje verbal
- $H_{0} :$ Proporción de visitas después del rediseño $\leq$ Proporción de visitas antes del rediseño
- $H_{1} :$ Proporción de visitas después del rediseño $>$ Proporción de visitas antes del rediseño

b.	Cargue la base de datos al entorno de trabajo.

In [1]:
import pandas as pd

df_9 = pd.read_csv("G5_A9_Compras_online.csv")
df_9.head()

,antes_rediseño,despues_rediseño
0,0,0.0
1,1,0.0
2,0,1.0
3,0,1.0
4,0,0.0


c.	A partir de la base de datos, obtenga la información necesaria para realizar el contraste de hipótesis.

In [2]:
import numpy as np
import scipy.stats as st

# Estadísticos muestrales necesarios
x1 = df_9["despues_rediseño"].sum()    # compras después del rediseño
x2 = df_9["antes_rediseño"].sum()      # compras antes del rediseño

n1 = df_9["despues_rediseño"].count()  # tamaño muestral después del rediseño
n2 = df_9["antes_rediseño"].count()    # tamaño muestral antes del rediseño

print(f"Después rediseño -> completa compra después rediseño = {x1}, n = {n1}")
print(f"Antes rediseño -> completa compra antes rediseño = {x2}, n = {n2}")


Después rediseño -> completa compra después rediseño = 90.0, n = 380
Antes rediseño -> completa compra antes rediseño = 68, n = 400


d.	Realice el contraste para la diferencia de proporciones.

In [3]:
alpha = 0.05

# Proporciones muestrales
p1 = x1 / n1
p2 = x2 / n2

# Proporción combinada
p_comb = (x1 + x2) / (n1 + n2)

# Estadístico de prueba
se = np.sqrt(p_comb * (1 - p_comb) * (1/n1 + 1/n2))
z_compras = (p1 - p2) / se

# Cálculo valor p
p_valor = 1 - st.norm.cdf(z_compras)

print(f"p1 (después) = {p1*100:.2f}%, p2 (antes) = {p2*100:.2f}%")
print(f"z = {z_compras:.3f}, p-valor = {p_valor:.4f}")

p1 (después) = 23.68%, p2 (antes) = 17.00%
z = 2.322, p-valor = 0.0101


In [4]:
# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

Rechazar H0


e.	¿Se puede afirmar, con un nivel de significancia del 5 %, que la proporción de visitas que terminan en compra es significativamente mayor después del rediseño que antes?

Con los datos analizados, la proporción de visitas que terminaron en compra fue de aproximadamente 23,7 % después del rediseño (90 de 380 visitas) y 17,0 % antes del rediseño (68 de 400 visitas). Al realizar el contraste de hipótesis para la diferencia de proporciones con un nivel de significancia del 5 %, se obtuvo un valor $p$ menor que $\alpha = 0,05$.

Por lo tanto, se rechaza la hipótesis nula. En conclusión, con la información disponible, es posible afirmar que la proporción de visitas que terminan en compra es significativamente mayor después del rediseño del sitio web.

# **Actividad 10 — Uso de transporte público entre dos campus**

Una universidad está impulsando políticas de sostenibilidad para reducir su huella de carbono. Una de las acciones clave es fomentar el uso de transporte público entre sus estudiantes en los distintos campus.

Para ello se implementaron estrategias distintas:

*	**Campus Norte:** además de la difusión habitual, se entregaron beneficios en la tarjeta de transporte (descuentos o recarga parcial) para motivar el uso del sistema público.
*	**Campus Sur:** solo se realizaron charlas informativas y campañas de sensibilización, sin beneficios económicos asociados.

Se dispone de un set de datos con información de estudiantes de ambos campus. El set de datos contiene las variables uso_transporte_norte y uso_transporte_sur, donde:

*	0 indica que el estudiante no usa transporte público.
*	1 indica que el estudiante sí usa transporte público.

A partir de esta información, determine si existe una diferencia en la proporción de estudiantes que usa transporte público entre el Campus Norte y el Campus Sur, utilizando un nivel de significancia del 5 %.
Descargue la base de datos **<a href="https://drive.google.com/uc?export=download&id=1DLR5JTkw08gN2_oo_8C_rd_zVkXXbHlN">aquí</a>.**


a. Formule las hipótesis del problema.

**Hipótesis**

- $H_{0} : p_{norte} = p_{sur}$
- $H_{1} : p_{norte} \neq p_{sur}$

En lenguaje verbal
- $H_{0} :$ Proporción estudiantes campus norte $=$ Proporción estudiantes campus sur
- $H_{1} :$ Proporción estudiantes campus norte $\neq$ Proporción estudiantes campus sur

b. Cargue la base de datos al entorno de trabajo.

In [5]:
import pandas as pd

df_10 = pd.read_csv("G5_A10_Transporte_publico.csv")
df_10.head()

,uso_transporte_norte,uso_transporte_sur
0,1,1.0
1,1,1.0
2,0,1.0
3,1,0.0
4,1,1.0


c. A partir de la base de datos, obtenga la información necesaria para realizar el contraste de hipótesis.

In [6]:
import numpy as np
import scipy.stats as st

# Estadísticos muestrales necesarios
x1 = df_10["uso_transporte_norte"].sum()    # uso transporte campus norte
x2 = df_10["uso_transporte_sur"].sum()      # uso transporte campus norte

n1 = df_10["uso_transporte_norte"].count()  # tamaño muestral del campus norte
n2 = df_10["uso_transporte_sur"].count()    # tamaño muestral del campus sur

print(f"Campus Norte -> utilizan transporte público campus norte = {x1}, n = {n1}")
print(f"Campus Sur -> utilizan transporte público campus sur = {x2}, n = {n2}")

Campus Norte -> utilizan transporte público campus norte = 160, n = 250
Campus Sur -> utilizan transporte público campus sur = 112.0, n = 200


d. Realice el contraste para la diferencia de proporciones.

In [7]:
import numpy as np
from scipy.stats import norm
import scipy.stats as st

alpha = 0.05

# Proporciones muestrales
p1 = x1 / n1
p2 = x2 / n2

# Proporción combinada
p_comb = (x1 + x2) / (n1 + n2)

# Estadístico de prueba
se = np.sqrt(p_comb * (1 - p_comb) * (1/n1 + 1/n2))
z_transporte = (p1 - p2) / se

# Cálculo valor p
p_valor = 2 * (1 - st.norm.cdf(abs(z_transporte)))

print(f"p1 (después) = {p1*100:.2f}%, p2 (antes) = {p2*100:.2f}%")
print(f"z = {z_transporte:.3f}, p-valor = {p_valor:.4f}")

p1 (después) = 64.00%, p2 (antes) = 56.00%
z = 1.725, p-valor = 0.0846


In [8]:
# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

No rechazar H0


e. ¿Se puede afirmar, con un nivel de significancia del 5 %, que la proporción de estudiantes que usa transporte público es diferente entre el Campus Norte y el Campus Sur?

Con los datos analizados, la proporción de estudiantes que utiliza transporte público es del 64 % en el Campus Norte (160 de 250 estudiantes) y del 56 % en el Campus Sur (112 de 200 estudiantes). Aunque la proporción es mayor en el Campus Norte, al realizar el contraste de hipótesis para la diferencia de proporciones con un nivel de significancia del 5 %, se obtuvo un valor $p$ mayor que 0,05.

Por lo tanto, no se rechaza la hipótesis nula. En conclusión, con la información disponible, no se puede afirmar que exista una diferencia estadísticamente significativa en la proporción de estudiantes que usa transporte público entre el Campus Norte y el Campus Sur.

# **Actividad 11 — Rendimiento de servidores**

Un equipo de desarrollo de software está evaluando el rendimiento de dos versiones de un mismo microservicio que procesa solicitudes de usuarios. La métrica clave es el tiempo de respuesta promedio (latencia en milisegundos).

- La versión A corresponde al servicio actual en producción.
- La versión B corresponde a una nueva versión con cambios en la arquitectura.

Antes de decidir si se mantiene la versión actual o se libera la nueva versión, el equipo desea comprobar si la latencia promedio del servicio actual es menor que la de la nueva versión, ya que un aumento en la latencia podría afectar negativamente la experiencia de usuario.

Para ello, se comparan dos muestras independientes de tiempos de respuesta, obtenidas bajo condiciones equivalentes de carga.

- Versión A: $n_{1}$ = 30, media = 198 ms, $s_{1}$ = 12  
- Versión B: $n_{2}$ = 30, media = 205 ms, $s_{2}$ = 10  

Con un nivel de significancia del 5 %, determine si existe evidencia estadísticamente suficiente para afirmar que la latencia promedio de la versión A es menor que la de la versión B. Luego de realizar el contraste de hipótesis correspondiente, ¿qué decisión debería tomar el equipo respecto del despliegue de la nueva versión?

1. Formule las hipótesis

**Hipótesis**

- $H_{0} : \mu_{versionA} >= \mu_{versionB}$

- $H_{1} : \mu_{versionA} < \mu_{versionB}$

En lenguaje verbal
- $H_{0} :$ Latencia promedio versión A $\geq$ Latencia promedio versión B

- $H_{1} :$ Latencia promedio versión A $<$ Latencia promedio versión B

2. Construya el constraste de hipótesis

In [ ]:
import numpy as np
from scipy import stats

# Datos
n1, x1, s1 = 30, 198, 12   # Versión A
n2, x2, s2 = 30, 205, 10   # Versión B
alpha = 0.05

# Error estándar
se = np.sqrt(s1**2/n1 + s2**2/n2)

# t de Welch (no asume varianzas iguales)
t_version = (x1 - x2) / se

# Grados de libertad
gl = (se**4) / (((s1**2/n1)**2)/(n1-1) + ((s2**2/n2)**2)/(n2-1))

# Cálculo valor p
p_valor = stats.t.cdf(t_version, df=gl)

# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(f"Estadístico de prueba t = {t_version:.3f}, Valor p = {p_valor:.4f}, Decisión = {decision}")

Estadístico de prueba t = -2.455, Valor p = 0.0086, Decisión = Rechazar H0


3. Interprete el resultado

Con $\alpha = 0,05$, el valor $p$ = 0,0086 es menor que 0,05. por lo tanto, se rechaza $H_{0}$, existe evidencia estadísticamente significativa para afirmar que la latencia promedio de la versión actual (A) es menor que la de la nueva versión (B), bajo las condiciones analizadas.

# **Actividad 12 — Participación de mujeres en ingeniería entre dos generaciones**

La Escuela de Ingeniería está impulsando acciones para aumentar la participación femenina en sus carreras, tradicionalmente masculinizadas. Como parte del seguimiento de estas iniciativas, desea comparar la presencia de mujeres entre una cohorte antigua y una cohorte reciente.

Se dispone de la siguiente información:

*	Cohorte 2015–2017: 300 estudiantes, de los cuales 45 son mujeres.
*	Cohorte 2022–2024: 320 estudiantes, de los cuales 70 son mujeres.

Con un nivel de significancia del 5 %, determine si existe evidencia estadísticamente suficiente para afirmar que la proporción de mujeres ha aumentado en la cohorte 2022–2024 respecto de la cohorte 2015–2017.

Luego de realizar el contraste de hipótesis correspondiente, ¿qué conclusión debería comunicar la escuela respecto del efecto de sus acciones de promoción?

1. Formule las hipótesis

**Hipótesis**

- $H_{0} : p_{2022-2024} \leq p_{2015-2017}$
- $H_{1} : p_{2022-2024} > p_{2015-2017}$

En lenguaje verbal
- $H_{0} :$ Proporción de mujeres cohorte 2022-2024 $\leq$ Proporción de mujeres cohorte 2015-2017
- $H_{1} :$ Proporción de mujeres cohorte 2022-2024 $>$ Proporción de mujeres cohorte 2015-2017

2. Construya el contraste de hipótesis

In [9]:
import numpy as np
from scipy.stats import norm

# Datos
n_rec, x_rec = 320, 70 # Cohorte 2015-2017
n_ant, x_ant = 300, 45 # Cohorte 2022-2024

# Proporciones muestrales
p_rec = x_rec / n_rec
p_ant = x_ant / n_ant

# Proporción combinada
p_comb = (x_rec + x_ant) / (n_rec + n_ant)

# Error estándar
se = np.sqrt(p_comb * (1 - p_comb) * (1/n_rec + 1/n_ant))

# Estadístico de prueba
z_mujeres = (p_rec - p_ant) / se

# Cálculo valor p
p_valor = 1 - norm.cdf(z_mujeres)

# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(f"p_rec = {p_rec*100:.2f}%, p_ant = {p_ant*100:.2f}%")
print(f"z = {z_mujeres:.3f}, p-valor = {p_valor:.4f}, Decisión = {decision}")

p_rec = 21.88%, p_ant = 15.00%
z = 2.201, p-valor = 0.0139, Decisión = Rechazar H0


3. Interprete el resultado

La participación femenina en la cohorte reciente (21,88 %) es mayor que en la cohorte antigua (15 %). Como el valor $p$ es menor que el nivel de significancia del 5 %, rechazamos $H_{0}$ y concluimos que hay evidencia estadística de que la proporción de mujeres en ingeniería ha aumentado significativamente en los últimos años.

# **Actividad 13 - Control de calidad: pureza del ácido en dos plantas**

Una empresa química desea comparar la pureza del ácido sulfúrico producido en dos plantas, ya que diferencias sistemáticas entre ellas podrían afectar los estándares de calidad del proceso.

Se registraron los siguientes porcentajes de pureza del ácido en muestras obtenidas desde cada planta:

*	Planta A: 96.8, 95.8, 96.2, 96.5, 96.4, 96.7, 96.3, 96.0, 97.1, 96.5
*	Planta B: 95.6, 96.0, 97.8, 98.4, 97.6, 98.2, 96.9, 96.8

Con un nivel de significancia del 1 %, determine si existe evidencia estadísticamente suficiente para afirmar que los porcentajes promedios de pureza del ácido difieren entre la Planta A y la Planta B.
Luego de realizar el contraste de hipótesis correspondiente, ¿qué conclusión operativa podría extraer la empresa respecto del control de calidad entre ambas plantas?

1. Formule las hipótesis

**Hipótesis**

- $H_{0} : \mu_{plantaA} = \mu_{plantaB}$

- $H_{1} : \mu_{plantaA} \neq \mu_{plantaB}$

En lenguaje verbal
- $H_{0} :$ Pureza promedio planta A $=$ Pureza promedio planta B

- $H_{1} :$ Pureza promedio planta A $\neq$ Pureza promedio planta B

2. Construya el constraste de hipótesis

In [ ]:
import numpy as np
from scipy.stats import t
import scipy.stats as st

# Datos
A = np.array([96.8, 95.8, 96.2, 96.5, 96.4, 96.7, 96.3, 96.0, 97.1, 96.5])
B = np.array([95.6, 96.0, 97.8, 98.4, 97.6, 98.2, 96.9, 96.8])
alpha = 0.01

# Estadísticos muestrales
n1, n2 = len(A), len(B)
x1, x2 = A.mean(), B.mean()
s1, s2 = A.std(ddof=1), B.std(ddof=1)

# Error estándar
se = np.sqrt(s1**2/n1 + s2**2/n2)

# t de Welch (no asume varianzas iguales)
t_pureza = (x1 - x2) / se

# Grados de libertad
gl = (se**4) / (((s1**2/n1)**2)/(n1-1) + ((s2**2/n2)**2)/(n2-1))

# Cálculo valor p
p_valor = 2 * (1 - t.cdf(abs(t_pureza), gl))

print(f"nA = {n1}, nB = {n2}")
print(f"mediaA = {x1:.3f}, mediaB = {x2:.3f}")
print(f"sA = {s1:.3f}, sB = {s2:.3f}")
print()
print(f"t = {t_pureza:.3f}, p-valor = {p_valor:.4f}")

nA = 10, nB = 8
mediaA = 96.430, mediaB = 97.162
sA = 0.383, sB = 1.014

t = -1.936, p-valor = 0.0864


In [ ]:
# Comparación y decisión
if p_valor < alpha:
    decision = "Rechazar H0"
else:
    decision = "No rechazar H0"

print(decision)

No rechazar H0


3. Interprete el resultado

Con $\alpha = 0,01$, el valor $p$ = 0,0864 es mayor que 0,01. por lo tanto, no se rechaza $H_{0}$. No hay evidencia suficiente al 1% para afirmar que los promedios de pureza del ácido sean diferentes entre la planta A y planta B.

# **Actividad 14 — Caso integrador**

Plantea tu propio escenario en tu área de interés (educación, salud, tecnología, medio ambiente, etc.) que compare dos proporciones. Por ejemplo, comparar tasas de aprobación en un examen antes y después de implementar un taller de preparación, tasas de satisfacción entre dos sedes, o cumplimiento de un indicador de calidad en dos períodos.

* Formula $H_{0}$ y $H_{1}$
* Indica tipo de prueba (izquierda, derecha o bilateral).
* Implementa el contraste en Python.
* Explica brevemente tu conclusión.

## En este laboratorio aprendiste a:

-	Formular y contrastar hipótesis estadísticas para la comparación entre dos poblaciones independientes, considerando tanto la diferencia de medias como la diferencia de proporciones, e identificando si corresponde una prueba unilateral o bilateral según el contexto del problema.
-	Desarrollar el procedimiento completo del contraste de hipótesis, interpretando el valor $p$ y la decisión estadística en función de un nivel de significancia previamente definido.
-	Analizar e interpretar los resultados obtenidos en contextos reales provenientes de ámbitos como la educación, la salud, la industria y la tecnología, reconociendo el rol de la estadística en la toma de decisiones basada en evidencia.
-	Aplicar en Python el procedimiento de contraste de hipótesis para comparar dos medias o dos proporciones, utilizando la información muestral disponible.

